pipeline/customer/customer.py

```diff
@@ -11,6 +11,69 @@
 from pipeline.customer import detail, order
 
 
+# pylint: disable=too-few-public-methods
+class Field(object):
+    CustomerId = 'customer_id'
+    Detail = 'detail'
+    Orders = 'orders'
+
+    Errors = 'errors'
+
+
+class Validate(beam.DoFn):
+
+    def process(self, element, *args, **kwargs):
+        customer_id, entry = element
+        details = list(entry.get(Field.Detail, []))
```

Comment:
Add a comment why you cast here (i.e. we get only an iterable but want to have a list)

Reply:
done